### PR TITLE
upgrade(package): Update Electron v1.6.6 -> v1.7.9

### PR DESCRIPTION
**Pull request by jhermsmeier**
This updates Electron from v1.6.6 to v1.7.9

Change-Type: patch
Changelog-Entry: Update Electron to v1.7.9

### CODE REVIEW

**Comment by lurch** [comment]
Forgive me for sounding nervous, but how well have you tested this @jhermsmeier ? ;-)  Given that we're planning to release "very soon" would it be safer to stay with 1.6.x and update to electron v1.6.15 ?  Or are there specific features in the 1.7.x line that we'd find useful too?

File: npm-shrinkwrap.json [hunk 1]
```diff
@@ -127,7 +133,7 @@
     },
     "angular-if-state": {
       "version": "1.0.0",
-      "from": "angular-if-state@>=1.0.0 <2.0.0",
+      "from": "angular-if-state@1.0.0",
```
Comment (lurch) [inline npm-shrinkwrap.json]: Travis is getting "upset" with so many changes in `npm-shrinkwrap.json` :-/  Can you check you're using the 'right' version of NPM?

Comment (lurch) [reply npm-shrinkwrap.json]: (and ping @jviotti too, in case he remembers how we resolved this problem the last time it cropped up)

Comment (jhermsmeier) [reply npm-shrinkwrap.json]: I was using the "right" npm version (which is node 6 with npm 3.10.x), the issue with this that the from field depends on the previous recorded version of the package, and will inevitably change when updating dependencies.

**Comment by jhermsmeier** [comment]
> Forgive me for sounding nervous, but how well have you tested this @jhermsmeier ?

I have, and everything appears to work just like it did before – but I'd encourage you to test it as well.

**Comment by jhermsmeier** [comment]
> would it be safer to stay with 1.6.x and update to electron v1.6.15 ? Or are there specific features in the 1.7.x line that we'd find useful too?

As we need up upgrade Electron due to the RCE issue anyways, and it's not breaking anything from what the tests and I can tell, I don't see why we should stick to the older 1.6.x – unless there are strong objections?

File: npm-shrinkwrap.json (hunk 1 above)
Comment (lurch) [reply npm-shrinkwrap.json]: :cry: 
But even so, why is the shrinkwrap file generated on Travis different to the shrinkwrap file in this PR? Or is it a horrible order-of-installation problem?

Comment (jhermsmeier) [reply npm-shrinkwrap.json]: So, I've just triple-checked – I'm on the same versions of node & npm as the travis box is, I shrinkwrapped multiple times, and no change – `make sanity-checks` comes back clean.

Comment (lurch) [reply npm-shrinkwrap.json]: CI servers seem happy now (and the shrinkwrap diff is much smaller, which is nice)
:-)

**Comment by lurch** [comment]
Tested it on Ubuntu 14.04 - tested the `npm start`, the electron-packaged version, the AppImage, and even (just for the hell of it) the CLI. All seem to work fine :-)

**Pull request merged**